### ISSUE TITLE
Subscript #1 of the array IDRY_E has value 4099 which is greater than the upper bound of 2684

### ISSUE THREAD

**Issue by platipodium** [schism-dev/schism-esmf#14]
@danishyo  Can you reproduce this with openmpmi device in our current test setup? https://pad.gwdg.de/KCGp_zhZR1af8ArudGljPw
 
```
Image              PC                Routine            Line        Source
schism_pdaf        000000000088FA13  Unknown               Unknown  Unknown
schism_pdaf        000000000042CE64  distribute_state_         181  distribute_state_pdaf.F90
schism_pdaf        000000000053FD46  Unknown               Unknown  Unknown
schism_pdaf        00000000004FC96B  schism_esmf_cap_m         858  schism_esmf_cap.F90
libesmf.so         00002AD3EE4ADF39  _ZN5ESMCI6FTable1     Unknown  Unknown
libesmf.so         00002AD3EE4B1F4B  ESMCI_FTableCallE     Unknown  Unknown
libesmf.so         00002AD3EEAE6801  _ZN5ESMCI3VMK5ent     Unknown  Unknown
libesmf.so         00002AD3EEAFC6A5  _ZN5ESMCI2VM5ente     Unknown  Unknown
libesmf.so         00002AD3EE4AF66A  c_esmc_ftablecall     Unknown  Unknown
libesmf.so         00002AD3EED682D1  esmf_compmod_mp_e     Unknown  Unknown
libesmf.so         00002AD3EEF93B5F  esmf_gridcompmod_     Unknown  Unknown
schism_pdaf        0000000000539AC7  MAIN__                    411  schism_pdaf.F90
schism_pdaf        00000000004120DE  Unknown               Unknown  Unknown
libc-2.17.so       00002AD3F22BD495  __libc_start_main     Unknown  Unknown
schism_pdaf        0000000000411FE9  Unknown               Unknown  Unknown
```

**Comment by platipodium** [comment]
This occurs inside SCHISM from the call `schism_step(it)` in schism_esmf_cap

**Comment by danishyo** [comment]
I do have similar error when using your ESMF build.
But when I try to use my own ESMF 8_2_0 build, this error is gone.
However, another error came out when finalizing ESMF.
See following:

20220606 141923.960 INFO             PET00 schism_001 ran.
20220606 141923.977 INFO             PET00 schism_001 finalizing ...
20220606 141923.977 ERROR            PET00 ESMF_StateAPI.cppF90:2565 ESMF_StateGet Not found  -  no ESMF_Field found named: wind_x-velocity_in_10m_height
20220606 141923.977 ERROR            PET00 schism_esmf_cap.F90:1111 Finalize Not found  - SCHISM subroutine call returned error
20220606 141923.977 INFO             PET00 Finalizing ESMF

**Comment by platipodium** [comment]
The latter  error 

```
20220606 141923.977 ERROR PET00 schism_esmf_cap.F90:1111 Finalize Not found - SCHISM subroutine call returned error
```

is now taken care of by commit df9d76d and prior.   

@danishyo 
(1) how do we proceed with the error in my build?
(2) can you go past this point with your build?

**Comment by danishyo** [comment]
> The latter error
> 
> ```
> 20220606 141923.977 ERROR PET00 schism_esmf_cap.F90:1111 Finalize Not found - SCHISM subroutine call returned error
> ```
> 
> is now taken care of by commit [df9d76d](https://github.com/schism-dev/schism-esmf/commit/df9d76d592c77ee24daf6e64e9459fa5d19b6e53) and prior.
> 
> @danishyo (1) how do we proceed with the error in my build? (2) can you go past this point with your build?

Thanks!
I re-test the code with my build, and now it pass with minor warning messages in PETs file.

20220606 173622.178 WARNING          PET01  ESMF_StateRemove Scalar itemName argument is deprecated.  Use itemNameList instead.

For (1), I don't have a good solution.
I notice your build is based on 8.3beta with debug mode.
Will this setting combine with beta version cause the effects?

**Comment by platipodium** [comment]
The warning should be gone now with commit 69f2bb1

Beta versions should not cause the problem.  I will try on other systems, as well.   Please confirm that the results are correct from your test simulations.

**Comment by danishyo** [comment]
Thanks!
I can get the correct result from 69f2bb1, but same warning messages still exist in PETs (which is OK for me).
ESMF 8.2.0 & 8.3.0 both works and deliver correct result without any problem.
I also notice, if debug option is on in ESMF build (ESMF_BOPT=g), same IDRY_E error will come back.
I will try Lavante later to see how it goes.

**Comment by platipodium** [comment]
this was a simple index error using `j` instead of `I`.  Fixed in c6d0faf
